x-pack/plugins/actions/kibana.json

```diff
@@ -4,7 +4,7 @@
   "version": "8.0.0",
   "kibanaVersion": "kibana",
   "configPath": ["xpack", "actions"],
-  "requiredPlugins": ["licensing", "taskManager", "encryptedSavedObjects", "eventLog", "features"],
-  "optionalPlugins": ["usageCollection", "spaces", "security"],
+  "requiredPlugins": ["licensing", "taskManager", "eventLog", "features"],
+  "optionalPlugins": ["usageCollection", "spaces", "security", "encryptedSavedObjects"],
```

Comment:
What's the expected behaviour when ESO is disabled?
Should Actions and Alerts still works without encryption?
If not, seems like this should still be in `requiredPlugins`.
If so, then seems like we're missing a whole test suite of e2e tests where Alerting & Actions run without ESOs

Reply:
> What's the expected behaviour when ESO is disabled?

In the alerting management and in the API responses we show the message that the user should configure ESO (like it was done for ephemeral encryption key. So I keep the same behavior.

Reply:
> Should Actions and Alerts still works without encryption?

Like it was said above, we are proposing to the user to enable the ESO plugin by the link `Learn how` in the UI message.

Reply:
> If so, then seems like we're missing a whole test suite of e2e tests where Alerting & Actions run without ESOs

Alerts and actions cannot run without ESO. 
ESO was in the optional plugins list, because the users should be able to get the info about possible missing configuration, I think..
Do you think we should disable it at all? 
To clarify: this PR changes is needed because there won't be support for ephemeral encryption key. But with this key, currently our framework is also not working for create and execute, but we are not disabling it and saying to the user that he should be able to change the configuration.

Reply:
I suspect that if we leave it as a required plugin, we'll end up with no alert/action UI at all, and the only diagnostic would be the list of plugins listed at Kibana startup, where alerts/actions wouldn't be listed.  Perhaps some messaging that the plugins were disabled because a required plugin wasn't available?  Or would Kibana just not start up at that point?

So seems odd, but the only reason to keep the alerts/actions plugins loadable is to display the message.  Which is probably important enough that we'd want to make this optional instead of required.  Presumably some read-only activities could still work (maybe not now, but could be _made_ to work?)?  Eg, listing alerts, alert instances?

Reply:
Actually, I see that `triggersActionsUi` already has `alerts` as an optional plugin: https://github.com/elastic/kibana/blob/28d61bf5fadcfaf3c051d2c20b3dab81b8dc0bcb/x-pack/plugins/triggers_actions_ui/kibana.json#L1-L7

So _in theory_ we could make ESO a required plugin of `alerts`, but then we won't **really** know why `alerts` wasn't loaded - I'm guessing there's not enough logging to make that determination while trying to diagnose the problem.  Could be ESO is not configured correctly.  Or something else.

So still sorta feels like `alerts` should make it optional ...

Reply:
In case if we left ESO as required plugin and set is as disabled, we will have Kibana started well, but the weird UI for our plugins will be like that:
<img width="1675" alt="Screen Shot 2021-01-13 at 12 42 58 PM" src="https://user-images.githubusercontent.com/55110838/104508395-03fbeb80-559d-11eb-87a4-a4af6a508d34.png">
Open a bug for this.

Actions and Alerts plugin both will be disabled and all API calls like `_health` return 404 NOT Found

Reply:
on master branch, if the setting `xpack.encryptedSavedObjects.enabled: false`:
<img width="941" alt="Screen Shot 2021-01-13 at 12 50 32 PM" src="https://user-images.githubusercontent.com/55110838/104509101-0c085b00-559e-11eb-94a3-c7a3c382606f.png">

Reply:
My personal opinion is that if ESO is required for alerts and actions to work then it should be a required plugin.
Our UI can add ESO as an optional plugin and if it's disabled show a suitable UI.

It seems to me that making it an optional plugin makes both the Alerts and Actions plugins more complicated than they need to be and makes it harder to maintain in the future because it makes it harder to reason about these plugins' "state".

I also think it's valid for the health endpoint to return 404 when the related plugin is disabled 🤷 

Perhaps we should have a sync chat with the whole team...

Reply:
Another option, BTW, is for TriggersActionsUI to have ESO as a required plugin at which point the UI doesn't appear at all when ESO is disabled.
As long as this is made clear in the documentation, I'd feel perfectly comfortable with that.

Reply:
> Perhaps we should have a sync chat with the whole team...

This is a great idea. Added to the discussion topics 👍

Reply:
🙏

Reply:
By the way f you decide to keep ESO as required dependency, then we may need to double-check with other solutions teams if they are still OK with keeping actions/alerts as required dependency or they want to switch to optional instead. I see that at least `case`, `infra`, `securitySolution` and `uptime` declare these plugins as required dependencies,

Reply:
@elastic/apm-ui, @elastic/siem, @elastic/uptime please let us know here if you have any objections about disabling alerts and actions plugins if encrypted_saved_objects is disabled.

Reply:
Will this impact if/when we move the alerting UI code to the `alerts` plugin?

Reply:
Good point. Yes, it will be disabled as well..

Reply:
I don't see how we have much of a choice here, this is required for things to be secure right? Is there any action we need to take on Uptime? If not, of course I support keeping things secure.

Reply:
@andrewvc just want to let you know that if we decide to make ESO as a required dependency for alerts and actions and uptime app will keep the alerts as required plugin it will be disabled as well on Kibana start.